### DISCUSSION TITLE
Order Cancellation States

### DISCUSSION THREAD

**Discussion by BLR-0118** [ONDC-Official/protocol-network-extension#20] [General]
What should be the cancellation states for an order, fulfillment? Since an order has multiple fulfillments (which have granular states), it may be easier to cancel a fulfillment at different states than an order.

An order can have the following states:
Created
Accepted
In-progress
Completed
Cancelled

A fulfillment can have the following states:
Pending
Packed
Order-picked-up
Out-for-delivery
Order-delivered
RTO-Initiated
RTO-Delivered
Cancelled

**Comment by BLR-0118** [comment]
@aakasjayaraj @abaweja @batramanuj @mahoriR @mherle @ekanshKR @VidyaGF

**Comment by BLR-0118** [comment]
@ONDC-Official/participants

**Comment by aakasjayaraj** [comment]
While coming to a consensus on this, one point to keep in mind is that the buyer should be able to cancel an order after the seller marks some items as out of stock or decrease the available qty of any item in an already-placed order.

**Comment by aanay1** [comment]
Hi Folks, 
I am receiving notifications from GitHub about the updates. 
Am I required to participate, or is it by mistake?

Thought - We may want to design the cancellation approval process for pre-payment and for COD, the process can either end in cancellation denied or partial/ total refund.
